File: lib/core/models/geo_point.dart
import 'package:flutter/foundation.dart';

@immutable
class GeoPoint {
  const GeoPoint(this.lat, this.lng, {this.heading});

  final double lat;
  final double lng;

  /// 進行方向（度、真北基準）。取得できない場合はnull。
  /// 位置の同一性比較（==/hashCode）には含めない。
  final double? heading;

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is GeoPoint && lat == other.lat && lng == other.lng;

  @override
  int get hashCode => Object.hash(lat, lng);
}

File: packages/engine/src/models/geo-point.ts
// 移植元: lib/core/models/geo_point.dart

/// 緯度経度。[heading] は進行方向（度、真北基準）で、取得できない場合は null。
///
/// Dart 版は `==`/`hashCode` から [heading] を外している（位置の同一性に含めない）。
/// TypeScript には演算子多重定義が無く、vitest の `toEqual` は構造比較なので
/// [heading] も比較対象に入る。#384 の移植対象6ファイルは [heading] を一切使わない
/// ため両者は一致するが、[heading] を持つ点の等値比較を書くときはここが食い違う。
export class GeoPoint {
  constructor(
    readonly lat: number,
    readonly lng: number,
    readonly heading: number | null = null,
  ) {}
}
